app/models/messages/Data.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.messages
+
+import models.departureP5.DepartureMessageType
+import models.departureP5.DepartureMessageType._
+import play.api.libs.json.{__, Json, OWrites, Reads}
+
+case class Data(data: MessageData)
+
+object Data {
+
+  def reads(messageType: DepartureMessageType): Reads[Data] =
+    messageType match {
+      case DepartureNotification => (__ \ "body" \ "n1:CC015C").read[MessageData].map(Data.apply)
+      case AmendmentSubmitted    => (__ \ "body" \ "n1:CC013C").read[MessageData].map(Data.apply)
+    }
```

Comment:
A refactor idea would be to have a 'toString' like method on DepartureMessageType and append the path e.g

(__ \ "body" \  s"n1:${messageType.toString}").read[MessageData].map(Data.apply)

Also we will get a matcher error here. Maybe have a catch all that will fail the reads if we its a different matcher error